addon-test-support/helpers.js

```diff
@@ -1,27 +1,27 @@
 import Ember from 'ember';
 import wait from 'ember-test-helpers/wait';
 
-const { run, $, merge } = Ember;
+const { run, merge } = Ember;
 
 const DEFAULT_EVENT_OPTIONS = { canBubble: true, cancelable: true };
 const KEYBOARD_EVENT_TYPES = ['keydown', 'keypress', 'keyup'];
 const MOUSE_EVENT_TYPES = ['click', 'mousedown', 'mouseup', 'dblclick', 'mouseenter', 'mouseleave', 'mousemove', 'mouseout', 'mouseover'];
 
 function focus(el) {
   if (!el) { return; }
-  let $el = $(el);
-  if ($el.is(':input, [contenteditable=true]')) {
-    let type = $el.prop('type');
+
+  if (el.tagName === 'INPUT' || el.contentEditable || el.tagName === 'ANCHOR') {
+    let type = el.type;
     if (type !== 'checkbox' && type !== 'radio' && type !== 'hidden') {
       run(null, function() {
         // Firefox does not trigger the `focusin` event if the window
-        // does not have focus. If the document doesn't have focus just
-        // use trigger('focusin') instead.
-
-        if (!document.hasFocus || document.hasFocus()) {
-          el.focus();
+        // does not have focus. If the document does not have focus then
+        // fire `focusin` event as well.
+        if (document.hasFocus && !document.hasFocus()) {
```

Comment:
@rwjblue I'd like to propose the same change to https://github.com/emberjs/ember.js/blob/aeeddf7c00b42a340494465252481b5414d87009/packages/ember-testing/lib/events.js#L8-L27 as Firefox can fail to use focus when the window is not in focus.